### PR TITLE
Use transport client and cleanup all the clients

### PR DESCRIPTION
**Pull request by bryanchriswhite**
https://storjlabs.atlassian.net/browse/V3-583

### CODE REVIEW

File: pkg/piecestore/rpc/client/client.go [hunk 1]
```diff
@@ -84,27 +89,33 @@ func NewCustomRoute(route pb.PieceStoreRoutesClient, nodeID *node.ID, bandwidthM
...
 	}, nil
 }
 
 // Close closes the connection with piecestore
-func (client *Client) Close() error {
-	return client.conn.Close()
+func (ps *Piecestore) Close() error {
+	if ps.closeFunc == nil {
+		// TODO@bryanchrswhite/aleitner: 	This will happen if someone uses `NewCustomRoute`.
+		// 																What is the expected behavior?
```
Comment (kaloyan-raev) [inline pkg/piecestore/rpc/client/client.go]: `NewCustomRoute` is used by the tests. So in this case the connection would be mocked, so returning just `nil` would be fine.

File: pkg/storage/ec/client.go [hunk 2]
```diff
@@ -28,45 +27,37 @@ var mon = monkit.Package()
...
-
-	conn, err := dialer.transport.DialNode(ctx, storageNode)
-	if err != nil {
-		return nil, err
+// NewClient from the given identity and max buffer memory
+func NewClient(identity *provider.FullIdentity, mbm int) Client {
+	tc := transport.NewClient(identity)
+	return &ecClient{
+		transport:       tc,
+		maxBuffMem:      mbm,
```
Comment (kaloyan-raev) [inline pkg/storage/ec/client.go]: If we changing the field name in the struct then lets change the `mbm` param in the constructor to `maxBufMem` too. Also, the short for "buffer" is usually "buf", not "buff".

File: pkg/audit/verifier.go [hunk 3]
```diff
@@ -57,26 +57,18 @@ func NewVerifier(transport transport.Client, overlay overlay.Client, id provider
...
-	if err != nil {
-		return nil, err
-	}
-	return client.NewPSClient(c, node.IDFromString(storageNode.GetId()), 0, d.identity.Key)
-}
-
 // getShare use piece store clients to download shares from a given node
 func (d *defaultDownloader) getShare(ctx context.Context, stripeIndex, shareSize, pieceNumber int,
-	id client.PieceID, pieceSize int64, node *pb.Node, authorization *pb.SignedMessage) (s share, err error) {
+	id psclient.PieceID, pieceSize int64, _node *pb.Node, authorization *pb.SignedMessage) (s share, err error) {
```
[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/audit/verifier.go]: Aliasing of package names is frowned upon because it makes it more difficult to understand things when moving from package to package. Consistency is key.

[the remark this answers is not in the document]
Comment (kaloyan-raev) [reply pkg/audit/verifier.go]: OK. I see. What's your opinion about `fromNode` vs `_node`? Are you fine using `_node` as param/variable names?

[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/audit/verifier.go]: Just `from` would do, but `fromNode` is fine as well.

File: pkg/piecestore/rpc/client/client.go [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright (C) 2018 Storj Labs, Inc.
 // See LICENSE for copying information.
 
-package client
+package psclient
```
Comment (egonelbre) [inline pkg/piecestore/rpc/client/client.go]: Let's move the package and rename the folder while were at it. Let's make it `pkg/piecestore/psclient` and `pkg/piecestore/psserver`. Can be a separate PR if you think this would get too confusing.

File: pkg/piecestore/rpc/client/client.go [hunk 5]
```diff
@@ -46,17 +46,22 @@ type PSClient interface {
 	io.Closer
 }
 
-// Client -- Struct Info needed for protobuf api calls
-type Client struct {
-	route            pb.PieceStoreRoutesClient // Client for interacting with Storage Node
-	conn             *grpc.ClientConn          // Connection to Storage Node
+// Piecestore -- Struct Info needed for protobuf api calls
+type Piecestore struct {
```
[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/piecestore/rpc/client/client.go]: I think `Client` was a good name here, I think the problem is more the interface location.

File: pkg/piecestore/rpc/client/pieceranger_test.go [hunk 6]
```diff
@@ -76,7 +75,14 @@ func TestPieceRanger(t *testing.T) {
 
 		ctx := context.Background()
 
-		c, err := NewCustomRoute(route, node.IDFromString("test-node-id-1234567"), 32*1024, priv)
+		_node := &pb.Node{
```
Comment (egonelbre) [inline pkg/piecestore/rpc/client/pieceranger_test.go]: Use `node` or `nodeInfo` or similar rather than the underscore prefix.

File: pkg/piecestore/rpc/client/pieceranger_test.go [hunk 7]
```diff
@@ -147,7 +153,14 @@ func TestPieceRangerSize(t *testing.T) {
...
```
Comment (egonelbre) [inline pkg/piecestore/rpc/client/pieceranger_test.go]: Use `node` or `nodeInfo` or `target` or `destination` rather than the underscore prefix.

File: pkg/pointerdb/pdbclient/client.go [hunk 8]
```diff
@@ -65,40 +66,34 @@ type Client interface {
...
-	if err != nil {
-		return nil, err
-	}
-
 	signatureHeader := &metadata.MD{}
-	peer := &peer.Peer{}
-	apiKeyInjector := grpcauth.NewAPIKeyInjector(APIKey, grpc.Header(signatureHeader), grpc.Peer(peer))
-	c, err := clientConnection(address, dialOpt, grpc.WithUnaryInterceptor(apiKeyInjector))
-
+	_peer := &peer.Peer{}
```
Comment (egonelbre) [inline pkg/pointerdb/pdbclient/client.go]: `peer` here was fine.

File: pkg/storage/ec/client.go (hunk 2 above)
Comment (egonelbre) [reply pkg/storage/ec/client.go]: `maxBufferSize`

Comment (kaloyan-raev) [reply pkg/storage/ec/client.go]: This value is not just about a single buffer, it's the max memory that all piece buffers can allocate together per single upload or download operation.

Comment (egonelbre) [reply pkg/storage/ec/client.go]: `memoryLimit` then?

File: examples/piecestore-client/main.go [hunk 9]
```diff
@@ -38,25 +36,21 @@ func main() {
...
-	if err != nil {
-		log.Fatal(err)
-	}
 
 	// Set up connection with rpc server
-	var conn *grpc.ClientConn
-	conn, err = grpc.Dial(":7777", identOpt)
-	if err != nil {
-		log.Fatalf("did not connect: %s", err)
+	_node := &pb.Node{
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply examples/piecestore-client/main.go]: `node` collides with a package name

File: pkg/pointerdb/pdbclient/client.go (hunk 8 above)
Comment (bryanchriswhite) [reply pkg/pointerdb/pdbclient/client.go]: `peer` is another package name from grpc

File: pkg/storage/ec/client.go (hunk 2 above)
Comment (kaloyan-raev) [reply pkg/storage/ec/client.go]: `memoryLimit` sounds good.

File: pkg/pointerdb/pdbclient/client.go (hunk 8 above)
Comment (bryanchriswhite) [reply pkg/pointerdb/pdbclient/client.go]: this is another package name collision

Comment (egonelbre) [reply pkg/pointerdb/pdbclient/client.go]: I'm not following. I guess the main nit is about avoiding underscores in front of variables, it doesn't look nice. I think having a variable `peer` here didn't cause problems, unless I missed something. Of course, there's possibility to use different name altogether, e.g. `peerInfo` or `peerAddress` etc.

Comment (mniewrzal) [reply pkg/pointerdb/pdbclient/client.go]: This part (`peer` variable) is removed in current master, no need for discussion :)

Comment (bryanchriswhite) [reply pkg/pointerdb/pdbclient/client.go]: the var `peer` would collide with [grpc's `peer`](https://godoc.org/google.golang.org/grpc/peer) package: `peer := &peer.Peer{}`

Comment (bryanchriswhite) [reply pkg/pointerdb/pdbclient/client.go]: It didn't end up mattering because `peer` was removed from the struct and the logic.
